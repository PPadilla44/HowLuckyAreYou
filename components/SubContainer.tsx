import { StyleProp, StyleSheet, Text, TextStyle, View } from 'react-native'
import React, { FC } from 'react'

interface Props {
    textStyle: StyleProp<TextStyle>
    title: string,
    text: string
}

const SubContainer: FC<Props> = ({ textStyle, title, text }) => {
    return (
        <View style={styles.subContainer}>
            <Text style={styles.title}>{title}</Text>
            <Text style={textStyle}>{text}</Text>
        </View>
    )
}

export default SubContainer

const styles = StyleSheet.create({
    subContainer: {
        alignItems: "center",
        justifyContent: "center",
        marginVertical: 70
    },
    title: {
        color: "#FBF8F1",
        fontWeight: "bold",
        fontSize: 24,
        textAlign: "center",
        fontFamily: "LexendDeca_700Bold"
        
    },
})