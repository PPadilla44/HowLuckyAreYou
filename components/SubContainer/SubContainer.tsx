import { StyleProp, StyleSheet, TextStyle,  } from 'react-native'
import { View, Text } from "../Themed"
import React, { FC } from 'react'

interface Props {
    textStyle: StyleProp<TextStyle>
    title: string,
    text: string
}

const SubContainer: FC<Props> = ({ textStyle, title, text }) => {
    return (
        <View style={styles.subContainer}>
            <Text style={styles.title} darkColor="#FBF8F1" lightColor='#004552' >{title}</Text>
            <Text style={textStyle}>{text}</Text>
        </View>
    )
}

export default SubContainer

const styles = StyleSheet.create({
    subContainer: {
        alignItems: "center",
        justifyContent: "center",
        marginVertical: 50
    },
    title: {
        fontWeight: "bold",
        fontSize: 24,
        textAlign: "center",
        fontFamily: "LexendDeca_700Bold"
    },
})