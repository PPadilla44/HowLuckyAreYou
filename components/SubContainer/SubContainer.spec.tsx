import { cleanup, render, RenderAPI } from "@testing-library/react-native";
import SubContainer, { Props } from "./SubContainer";

describe('<SubContainer/>', () => {

    const props: Props = {
        text: "TEST TEXT",
        textStyle: {
            fontWeight: "bold",
            fontSize: 64,
            fontFamily: "Futura"
        },
        title: "TEST TITLE"
    }

    afterEach(cleanup)

    let wrapper: RenderAPI;
    beforeEach(() => {
        wrapper = render(<SubContainer {...props} />);
    })

    it("renders props", () => {
        expect(wrapper.container.props).toEqual(props);
    });

    it("renders texts", () => {
        const title = wrapper.getByTestId("title");
        expect(title.props.children).toBe("TEST TITLE");
        
        const text = wrapper.getByTestId("text")
        expect(text.props.children).toBe("TEST TEXT");
    });
    
    it("render style from text style props", () => {
        const text = wrapper.getByTestId("text");
        expect(text).toHaveStyle({
            fontWeight: "bold",
            fontSize: 64,
            fontFamily: "Futura"
        });
        
    });
    
    it("renders correctly", () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
    
    it("updates correctly", () => {
        const newComponent = (
            <SubContainer 
            text="new text"
            title="new title"
            textStyle={{
                backgroundColor: "red",
            }} 
            />
        )
        wrapper.update(newComponent)
        expect(wrapper.toJSON()).toMatchSnapshot()
    })
    
})