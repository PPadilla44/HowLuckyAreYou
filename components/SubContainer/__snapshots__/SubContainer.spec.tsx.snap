// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubContainer/> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "shadowColor": "white",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "space-between",
      },
    ]
  }
  testID="SubContainer"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Futura",
        },
        Object {
          "color": "#325C99",
          "fontFamily": "Futura",
          "fontSize": 24,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
    testID="title"
  >
    TEST TITLE
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Futura",
        },
        Object {
          "fontFamily": "Futura",
          "fontSize": 64,
          "fontWeight": "bold",
        },
      ]
    }
    testID="text"
  >
    TEST TEXT
  </Text>
</View>
`;

exports[`<SubContainer/> updates correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "black",
        "shadowColor": "white",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "space-between",
      },
    ]
  }
  testID="SubContainer"
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Futura",
        },
        Object {
          "color": "#325C99",
          "fontFamily": "Futura",
          "fontSize": 24,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
    testID="title"
  >
    new title
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "Futura",
        },
        Object {
          "backgroundColor": "red",
        },
      ]
    }
    testID="text"
  >
    new text
  </Text>
</View>
`;
